src/loader.ts

```diff
@@ -215,18 +229,30 @@ async function extendConfig(config, options: LoadConfigOptions) {
     );
     delete config[key];
   }
-  for (const extendSource of extendSources) {
+  for (let extendSource of extendSources) {
+    const originalExtendSource = extendSource;
+    let sourceOptions = {};
+    if (extendSource.source) {
+      sourceOptions = extendSource.options || {};
+      extendSource = extendSource.source;
+    }
```

Comment:
I would remove this to only keep `[source, sourceOptions]`

Reply:
We might make this decition on framework level. So far this feature is not even typed for either of option syntaxes so we can decide for Nuxt. But generally i don't see a problem with supporting both syntaxes (and object syntax might more future proof API for supporting other context keys)